pkg/e2e/operators/cloudingress/public_private.go

```diff
@@ -2,35 +2,100 @@ package cloudingress
 
 import (
 	"context"
+	"log"
+	"time"
 
 	"github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
+	cloudingressv1alpha1 "github.com/openshift/cloud-ingress-operator/pkg/apis/cloudingress/v1alpha1"
 
 	"github.com/openshift/osde2e/pkg/common/constants"
 	"github.com/openshift/osde2e/pkg/common/helper"
 
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
 )
 
 // tests
 var _ = ginkgo.Describe(constants.SuiteInforming+TestPrefix, func() {
 	h := helper.New()
 
 	ginkgo.Context("publishingstrategy-public-private", func() {
-		ginkgo.It("is a placeholder", func() {
-			_, _ = h.Kube().CoreV1().Pods(TestPrefix).Get(context.TODO(), "something", metav1.GetOptions{})
+		ginkgo.It("should be able to toggle the default applicationingress from public to private", func() {
 
-			tests := []struct {
-				Name string
-			}{}
+			updateApplicationIngress(h, "internal")
 
-			for _, test := range tests {
-				Expect(test.Name).To(Equal(""))
-			}
+			//wait for router-default service loadbalancer to have an annotation indicating its scheme is internal
+			err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
```

Comment:
Is 5 mins the right deadline for this? What are the changes of this being racy with the 5 min deadline?

Reply:
While testing this, I never saw it take anywhere near that long, generally less than a minute. Most of that time is spent waiting for the ingress pods to be terminated/recreated. I think its unlikely to be racy.